Remotion/Validation/Core/Validators/InclusiveBetweenValidator.cs

```diff
@@ -1,33 +1,58 @@
-﻿// Decompiled with JetBrains decompiler
-// Type: FluentValidation.Validators.InclusiveBetweenValidator
-// Assembly: FluentValidation, Version=5.0.0.1, Culture=neutral, PublicKeyToken=a82054b837897c66
-// MVID: 30628A95-CE3F-41E4-BA2A-29882CBD79CE
-// Assembly location: C:\Development\Remotion\trunk-svn2git\packages\FluentValidation-Signed.5.0.0.1\lib\Net40\FluentValidation.dll
-
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
 using System;
+using JetBrains.Annotations;
+using Remotion.Utilities;
 using Remotion.Validation.Implementation;
 
 namespace Remotion.Validation.Validators
 {
   public class InclusiveBetweenValidator : PropertyValidator, IBetweenValidator
   {
-    public InclusiveBetweenValidator (IComparable from, IComparable to, IValidationMessage validationMessage = null)
+    public IComparable From { get; }
+    public IComparable To { get; }
+
+    public InclusiveBetweenValidator ([NotNull] IComparable from, [NotNull] IComparable to, [CanBeNull] IValidationMessage validationMessage = null)
         : base (Constants.InclusiveBetweenError, validationMessage ?? new InvariantValidationMessage (Constants.InclusiveBetweenError))
     {
+      ArgumentUtility.CheckNotNull ("from", from);
+      ArgumentUtility.CheckNotNull ("to", to);
+
       To = to;
       From = from;
-      if (Comparer.GetComparisonResult (to, from) == -1)
-        throw new ArgumentOutOfRangeException (nameof (to), "To should be larger than from.");
-    }
 
-    public IComparable From { get; }
+      if (From.GetType() != To.GetType())
+        throw new ArgumentException ("From must have the same type as to.", "to");
 
-    public IComparable To { get; }
+      if (To.CompareTo (from) == -1)
+        throw new ArgumentOutOfRangeException ("to", "To should be larger than from.");
```

Comment:
not readlly sure about precedence, but I think I would like to have 'to' and 'from' recognizable in the message. maybe 'to' and 'from' or to-value and from-value.?